src/services/codefixes/fixAddMissingParam.ts

```diff
@@ -0,0 +1,248 @@
+
+        const changes = textChanges.ChangeTracker.with(context, t => doChange(t, context.sourceFile, info));
+        return [createCodeFixAction(fixId, changes, info.description, fixId, Diagnostics.Add_all_missing_parameters)];
+    },
+    getAllCodeActions: context =>
+        codeFixAll(context, errorCodes, (changes, diag) => {
+            const info = getInfo(context.sourceFile, context.program, diag.start);
+            if (info) {
+                doChange(changes, context.sourceFile, info);
+            }
+        }),
+});
+
+type ConvertableSignatureDeclaration =
+    | FunctionDeclaration
+    | FunctionExpression
+    | ArrowFunction
+    | MethodDeclaration;
+
+interface SignatureInfo {
+    readonly description: DiagnosticOrDiagnosticAndArguments;
+    readonly newParameters: ParameterInfo[];
+    readonly declarations: ConvertableSignatureDeclaration[];
+    readonly overload: ConvertableSignatureDeclaration | undefined;
+}
+
+interface ParameterInfo {
+    readonly pos: number;
+    readonly declaration: ParameterDeclaration;
+}
+
+function getInfo(sourceFile: SourceFile, program: Program, pos: number): SignatureInfo | undefined {
+    const token = getTokenAtPosition(sourceFile, pos);
+    const callExpression = findAncestor(token, isCallExpression);
+    if (callExpression === undefined || length(callExpression.arguments) === 0) return undefined;
+
+    const checker = program.getTypeChecker();
+    const type = checker.getTypeAtLocation(callExpression.expression);
+    const signatures = checker.getSignaturesOfType(type, SignatureKind.Call);
+    const signature = lastOrUndefined(sort(signatures, (a, b) => length(a.parameters) - length(b.parameters)));
+    if (
+        signature &&
+        signature.declaration &&
+        isConvertableSignatureDeclaration(signature.declaration)
+    ) {
+        const declaration = (signature.declaration.body === undefined ? find(signature.declaration.symbol.declarations, d => isConvertableSignatureDeclaration(d) && !!d.body) : signature.declaration) as ConvertableSignatureDeclaration;
+        if (declaration === undefined) {
+            return undefined;
+        }
+
+        isSourceFileFromLibrary(program, declaration.getSourceFile());
+
+        const overload = signature.declaration.body === undefined ? signature.declaration : undefined;
+        if (overload && length(overload.parameters) !== length(declaration.parameters)) return undefined;
+
+        const name = tryGetName(declaration);
+        if (name === undefined) return undefined;
+
+        const declarations: ConvertableSignatureDeclaration[] = append([declaration], overload);
+        const newParameters: ParameterInfo[] = [];
+        const parametersLength = length(declaration.parameters);
+        const argumentsLength = length(callExpression.arguments);
+        if (parametersLength > argumentsLength) return undefined;
+
+        for (let i = 0, pos = 0, paramIndex = 0; i < argumentsLength; i++) {

```

Comment:
it's a little unfortunate that this code is, essentially, re-implementing the parameter-argument assignability checks in resolveCall. But it's probably the right approach, since exposing resolveCall for re-use isn't feasible.